src/main/java/org/jenkinsci/plugins/gitclient/LegacyCompatibleGitAPIImpl.java

```diff
@@ -195,7 +195,10 @@ public void clone(RemoteConfig rc, boolean useShallowClone) throws GitException,
     /** {@inheritDoc} */
     @Override
     protected Object writeReplace() {
-        return remoteProxyFor(Channel.current().export(IGitAPI.class, this));
+        Channel currentChannel = Channel.current();
+        if (currentChannel == null)
+            return null;
```

Comment:
I would throw https://docs.oracle.com/javase/7/docs/api/java/io/ObjectStreamException.html or so to block serialization over other channels at all and to propagate the error